### PR TITLE
fix(perf): Dedupe component risk upserts

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

Image and component risk factors are not actually used currently so only insert the values if the score has changed, which will reduce the number of writes on the database

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Scale test + comparison of QA metrics

Before 

cat stackrox_debug_2023_10_24_06_48_08/metrics-1 | grep postgres_op_duration_count | grep Risk
rox_central_postgres_op_duration_count{Operation="Get",Type="Risk"} 21616
rox_central_postgres_op_duration_count{Operation="Remove",Type="Risk"} 3842
rox_central_postgres_op_duration_count{Operation="Upsert",Type="Risk"} 10811

After:
cat stackrox_debug_2023_10_24_17_43_27/metrics-1 | grep postgres_op_duration_count | grep Risk
rox_central_postgres_op_duration_count{Operation="Get",Type="Risk"} 21136
rox_central_postgres_op_duration_count{Operation="Remove",Type="Risk"} 3711
rox_central_postgres_op_duration_count{Operation="Upsert",Type="Risk"} 5615

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/risk/manager/manager.go [hunk 1]
```diff
@@ -241,14 +241,16 @@ func (e *managerImpl) reprocessImageComponentRisk(imageComponent *storage.Embedd
...
-		log.Errorf("Error reprocessing risk for image component %s v%s: %v", imageComponent.GetName(), imageComponent.GetVersion(), err)
-	}
 
+	// Image risk results are not currently used so if the score is the same then no need to upsert
 	if oldScore == risk.GetScore() {
 		return
 	}
 
+	if err := e.riskStorage.UpsertRisk(riskReprocessorCtx, risk); err != nil {
+		log.Errorf("Error reprocessing risk for image component %s v%s: %v", imageComponent.GetName(), imageComponent.GetVersion(), err)
```
Comment (md2119) [inline central/risk/manager/manager.go:251]: We should not append `v` to the version. The end user will likely include it in their searches consequently the search will fail.

File: central/risk/manager/manager.go [hunk 2]
```diff
@@ -263,6 +265,14 @@ func (e *managerImpl) reprocessNodeComponentRisk(nodeComponent *storage.Embedded
 		return
 	}
 
+	oldScore := e.nodeComponentRanker.GetScoreForID(
+		scancomponent.ComponentID(nodeComponent.GetName(), nodeComponent.GetVersion(), os))
+
+	// Node risk results are not currently used so if the score is the same then no need to upsert
+	if oldScore == risk.GetScore() {
+		return
+	}
```
Comment (md2119) [inline central/risk/manager/manager.go:274]: Flip the order.

Comment (connorgorman) [reply central/risk/manager/manager.go:274]: Not sure I'm following on the order flip?

Comment (md2119) [reply central/risk/manager/manager.go:274]: Ah, `GetScoreForID` is not a datastore op. I misread it.

File: central/risk/manager/manager.go (hunk 1 above)
Comment (connorgorman) [reply central/risk/manager/manager.go:251]: Will remove

**Review by md2119 — APPROVED** [review]

**Pull request merged**